octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/util/EntityComboBox.java

```diff
@@ -133,73 +133,115 @@ public EntityComboBox(Composite parent, int style, LabelProvider labelProvider,
-                Display.getDefault().syncExec(() -> {
-                    if(optionsComposite.isDisposed()) {
-                        return;
-                    }
-                    Arrays.stream(optionsComposite.getChildren()).forEach(Control::dispose);
-                    
-                    LoadingComposite loadingComposite = new LoadingComposite(optionsComposite, SWT.NONE);
-                    loadingComposite.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
-                    optionsComposite.setSize(optionsComposite.getParent().getSize());
-                    optionsComposite.layout();
-                });
+                if (!rootComposite.isDisposed()) {
+                    showLoading();
+                }
             }
+
             @Override
             public void done(IJobChangeEvent event) {
+                IStatus status = event.getResult();
+
                 Display.getDefault().syncExec(() -> {
-                    
-                    //The shell might have been close before the result has been fetched from the server
-                    //In this case discard the result
-                    if(optionsComposite.isDisposed()) {
+
+                    // The shell might have been close before the result has
+                    // been fetched from the server
+                    // In this case discard the result
+                    if (rootComposite.isDisposed()) {
                         return;
                     }
-                    
-                    Arrays.stream(optionsComposite.getChildren()).forEach(Control::dispose);
-                    
-                    if(entityList == null || entityList.isEmpty()) {
-                        Label lblNoResult = new Label(optionsComposite, SWT.NONE);
-                        lblNoResult.setText("No results");
-                        lblNoResult.setLayoutData(new GridData(SWT.CENTER, SWT.CENTER, true, true));
-                    } else if(selectionMode == SWT.SINGLE){
-                        createSingleSelectionUI();
-                    } else if(selectionMode == SWT.MULTI){
-                        createMultiSelectionUI();
+
+                    if (Status.OK_STATUS == status) {
+                        Arrays.stream(rootComposite.getChildren()).forEach(Control::dispose);
+
+                        if (entities == null || entities.isEmpty()) {
+                            Label lblNoResult = new Label(rootComposite, SWT.NONE);
+                            lblNoResult.setText("No results");
+                            lblNoResult.setLayoutData(new GridData(SWT.CENTER, SWT.CENTER, true, true));
+                        } else if (selectionMode == SWT.SINGLE) {
+                            showSingleSelectionUI();
+                        } else if (selectionMode == SWT.MULTI) {
+                            showMultiSelectionUI();
+                        }
+
+                    } else if (Status.WARNING == status.getSeverity()) {
```

Comment:
@atothhpe Is it ok to use getSeverity() and above just status? What about the cancel or error cases?

Reply:
Cancel and error cases can't happen unless you return them explicitly in the job method